stream.go

```diff
@@ -91,8 +81,8 @@ func (s *Stream) waitForData(ctx context.Context) error {
 		s.extra = read
 		s.exbuf = read
 		return nil
-	case <-ctx.Done():
-		return ctx.Err()
+	case <-s.rDeadline.wait():
+		return context.DeadlineExceeded
```

Comment:
Should probably return a `timeout` error instead.

Reply:
good point, will do.

Reply:
done